portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,30 +58,60 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         if(comments.size() < numComments){
           long id = entity.getKey().getId();
           String text = (String) entity.getProperty("text");
-          long timestamp = (long) entity.getProperty("timestamp");
 
-          Comment comment = new Comment(id,text,timestamp);
+          // Do the translation.
+          Translate translate = TranslateOptions.getDefaultInstance().getService();
+          Translation translation =
+            translate.translate(text, Translate.TranslateOption.targetLanguage(language));
+          String translatedText = translation.getTranslatedText();
+          
+          long timestamp = (long) entity.getProperty("timestamp");
+          String email = (String) entity.getProperty("email");
+          String nickname= (String) entity.getProperty("nickname");
+          Comment comment = new Comment(id,translatedText,timestamp,email, nickname);
           comments.add(comment);
         }
         else{
             break;
         } 
     }
     Gson gson = new Gson();
-    response.setContentType("application/json;");
+    response.setContentType("application/json");
+    response.setCharacterEncoding("UTF-8");
     response.getWriter().println(gson.toJson(comments));
+    
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String text = request.getParameter("text");
     long timestamp = System.currentTimeMillis();
+    String email = userService.getCurrentUser().getEmail();
```

Comment:
Is the user service guaranteed to return an email? Can it ever return null or an empty email address? Your code should handle these cases if they are possible failure modes.

Reply:
Yes, because in authentication you need to be signed into your email before posting a comment.